src/webui/webapplication.cpp

```diff
@@ -533,11 +533,14 @@ Http::Response WebApplication::processRequest(const Http::Request &request, cons
     header(Http::HEADER_X_XSS_PROTECTION, "1; mode=block");
     header(Http::HEADER_X_CONTENT_TYPE_OPTIONS, "nosniff");
 
+    QString csp = QLatin1String("default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; script-src 'self' 'unsafe-inline'; object-src 'none';");
     if (m_isClickjackingProtectionEnabled) {
         header(Http::HEADER_X_FRAME_OPTIONS, "SAMEORIGIN");
-        header(Http::HEADER_CONTENT_SECURITY_POLICY, "default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; script-src 'self' 'unsafe-inline'; object-src 'none';");
+        csp += QLatin1String(" frame-ancestors 'self';");
     }
 
+    header(Http::HEADER_CONTENT_SECURITY_POLICY, csp);
```

Comment:
So you still think that the user may not be good reasons to disable it completely?

Reply:
yes, csp is useful no matter what.
And for the record: https://github.com/qbittorrent/qBittorrent/pull/8967#issuecomment-392644149